Extension/lib/content-script/preload.js

```diff
@@ -126,52 +126,84 @@
     };
 
     /**
-     * Overrides window.WebSocket running the function from websocket.js
+     * Try to keep DOM clean: let script removes itself when execution completes
+     * @returns {string}
+     */
+    var cleanupCurrentScriptToString = function () {
+
+        var cleanup = function () {
+            var current = document.currentScript;
+            var parent = current && current.parentNode;
+            if (parent) {
+                parent.removeChild(current);
+            }
+        };
+
+        return '(' + cleanup.toString() + ')();';
+    };
+
+    /**
+     * Overrides window.WebSocket and window.RTCPeerConnection running the function from wrappers.js
      * https://github.com/AdguardTeam/AdguardBrowserExtension/issues/203
+     * https://github.com/AdguardTeam/AdguardBrowserExtension/issues/588
      */
-    /* global initPageMessageListener, overrideWebSocket */
-    var initWebSocketWrapper = function () {
-        // This is chrome-specific feature for blocking WebSocket connections
-        // overrideWebSocket function is not defined in case of other browsers
-        if (typeof overrideWebSocket !== 'function') {
-            return;
-        }
+    /* global initPageMessageListener, overrideWebSocket, overrideWebRTC, injectPageScriptAPI */
+    var initRequestWrappers = function () {
 
         // Only for dynamically created frames and http/https documents.
         if (!isHtml() &&
             window.location.href !== "about:blank") {
             return;
         }
 
+        var userAgent = navigator.userAgent.toLowerCase();
+        if (userAgent.indexOf('firefox') >= 0) {
+            // Explicit check, we must not go further in case of Firefox

```

Comment:
It was AMO reviewer's requirement.
See task and commit: https://github.com/AdguardTeam/AdguardBrowserExtension/issues/379